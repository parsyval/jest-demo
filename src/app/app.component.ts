import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'app';
  isActive = false;
  
  activate(){
    this.isActive = !this.isActive;
  }

  getText(){
    return this.isActive ? "I'm triggered" : "I'm not triggered";
  }
}
