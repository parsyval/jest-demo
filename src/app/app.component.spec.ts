import { TestBed, async } from '@angular/core/testing';

import { AppComponent } from './app.component';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  }));

  it('should create the app', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  }));

  it(`should have as title 'app'`, async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app.title).toEqual('app');
  }));

  it('should render title in a h1 tag', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled.querySelector('h1').textContent).toContain('Welcome to app!');
  }));

  fdescribe('JEST TEST HTML', () => {
    it('Should be deactivated on init', () => {
      const fixture = TestBed.createComponent(AppComponent);
      fixture.detectChanges();
      const compiled = fixture.debugElement.nativeElement;

      expect(compiled.querySelector('#text').textContent).toContain('I\'m not triggered');
    });

    it('Should change when the button is clicked',async () => {
      const fixture = TestBed.createComponent(AppComponent);
      fixture.detectChanges();
      const compiled = fixture.debugElement.nativeElement;

      compiled.querySelector('#button').click();
      //console.log('button', compiled.querySelector('#button').click);
      fixture.detectChanges();
      await fixture.whenStable();

      expect(compiled.querySelector('#text').textContent).toMatchSnapshot();
    });
  });
});
