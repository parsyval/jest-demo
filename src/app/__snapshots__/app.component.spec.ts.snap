// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent JEST TEST HTML Should change when the button is clicked 1`] = `"I'm triggered"`;
